code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/ServiceCreateValidationFilter.java

```diff
@@ -71,6 +77,14 @@ public Object create(String type, ApiRequest request, ResourceManager next) {
         return super.create(type, request, next);
     }
 
+    protected void validateEnvironment(Service service) {
+        Environment env = objectManager.loadResource(Environment.class, service.getEnvironmentId());
+        if (StringUtils.equals(env.getState(), CommonStatesConstants.ERROR)) {
+            throw new ValidationErrorException(ValidationErrorCodes.INVALID_STATE, 
```

Comment:
@ibuildthecloud said - 

Is this an action? @alena1108 When does this code get called?

Then, @ibuildthecloud and I, while offline agreed to call this a new error called as INVALID_STATE.